doc/crdt-details.md

```diff
@@ -0,0 +1,1054 @@
+///
+/// Since only the deleted portions of the union string are necessary,
+/// instead of requiring a union string the function takes a `tombstones`
+/// rope which contains the deleted portions of the union string. The
+/// `from_dels` subset must be the interleaving of `tombstones` into the
+/// union string.
+pub fn synthesize(tombstones: &Node<N>, from_dels: &Subset, to_dels: &Subset) -> Delta<N> {
+  // ...
+}
+```
+
+### Engine::delta_rev_head
+
+This operation is similar to `Engine::get_head` except it returns a `Delta` from the text at a specified revision to the current head text. This is useful for things like updating the position of cursors and rich text spans when edits are made.
+
+Like `Engine::get_head` it starts by calling `Engine::deletes_from_cur_union_for_index` to get a `Subset` describing the state of the text at the old revision relative to the current union string. Now we can just use `Delta::synthesize` to create a `Delta` from the old to the new `deletes_from_union`. The problem is, `Delta::synthesize` expects the tombstones `Rope` you give it to correspond to `from_dels`, but we have one for `to_dels`. To fix this, we can use a helper called `shuffle_tombstones` to shuffle things to get an `old_tombstones` corresponding to `from_dels`.
+
+#### shuffle_tombstones
+
+Constructing a `tombstones` to go with an older `deletes_from_union` given the current `text`, `tombstones` and `deletes_from_union` may involve moving characters from `text` to the old `tombstones`, and removing characters from the `tombstones` that weren't deleted at the time. We can describe the necessary changes using a `Delta`, but how do we create it?
+
+We can note that since the 0-segments of `deletes_from_union` correspond to characters in `text` and the non-zero segments correspond to characters in `tombstones`, if we take the complement (0 becomes 1, non-zero becomes 0) of `deletes_from_union` we can use helpers designed for use with tombstones on `text` and vice-versa. So we can use `Delta::synthesize` with the `text` (where you would normally put the tombstones) and the complement of the old and new `deletes_from_union` to create the required `Delta` of the `tombstones` when it normally creates deltas of `text`.
+
+```rust
+/// Move sections from text to tombstones and out of tombstones based on a new and old set of deletions
+fn shuffle_tombstones(text: &Rope, tombstones: &Rope,
+        old_deletes_from_union: &Subset, new_deletes_from_union: &Subset) -> Rope {
+    // Taking the complement of deletes_from_union leads to an interleaving valid for swapped text and tombstones,
+    // allowing us to use the same method to insert the text into the tombstones.
+    let inverse_tombstones_map = old_deletes_from_union.complement();
+    let move_delta = Delta::synthesize(text, &inverse_tombstones_map, &new_deletes_from_union.complement());
+    move_delta.apply(tombstones)
+}
+```
+
+### Engine::edit_rev
```

Comment:
Another name nit: does `edit_rev` actually 'edit' a revision? or is it more just like `new_rev_based_on_another_previous_rev`?